src/Ethereum/Address.cpp

```diff
@@ -37,6 +37,9 @@ Address::Address(const std::vector<uint8_t>& data) {
 }
 
 Address::Address(const PublicKey& publicKey) {
+    if (publicKey.type() != PublicKeyType::secp256k1Extended) {
+        throw std::invalid_argument("Ethereum::Address needs an extended SECP256k1 public key.");
```

Comment:
I also wonder if we do this for Ethereum, should we do for bitcoin as well?

Reply:
done